baselines/flanders/README.md

```diff
@@ -0,0 +1,119 @@
+
+Income (binary classification):
+- cyclic coordinate descent (CCD)
+- L1-regularized binary cross-entropy loss (LASSO)
+
+MNIST (multilabel classification, fully connected, feed forward NN):
+- Multilevel Perceptron (MLP)
+- minimizing multiclass cross-entropy loss using Adam optimizer
+- input: 784
+- hidden layer 1: 128
+- hidden layer 2: 256
+
+
+****Dataset:**** Every dataset is partitioned into two disjoint sets: 80% for training and 20% for testing. The training set is distributed uniformly across all clients (100), while the testing set is held by the server to evaluate the global model.
+
+| Description | Default Value |
+| ----------- | ----- |
+| Partitions | 100 |
+| Evaluation | centralized |
+| Training set | 80% |
+| Testing set | 20% |
+
+****Training Hyperparameters:****
+
+| Dataset | # of clients  | Clients per round | # of rounds | $L$ | Batch size | Learning rate | $\lambda_1$ | $\lambda_2$ | Optimizer | Dropout | Alpha | Beta | # of clients to keep | Sampling |
+| -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- |
+| Income | 10 | 100 | 50 | 30 | \ | \ | 1.0 | 0.0 | CCD | \ | 0.0 | 0.0 | 1 | \ |
+| MNIST | 10 | 100 | 50 | 30 | 32 | $10^{-3}$ | \ | \ | Adam | 0.2 | 0.0 | 0.0 | 1 | \ |
+
+Where $\lambda_1$ and $\lambda_2$ are Lasso and Ridge regularization terms, $L$ is the number of warmup rounds (rounds without any attack) and the number of malicious clients ranges from 0 to 9.
+
+
+## Environment Setup
+
+```bash  
+# Use a version of Python >=3.9 and <3.12.0.
+poetry env use 3.9.18
+
+# Install everything from the toml
+poetry install
+
+# Activate the env
+poetry shell
+
+# Manually install natsort
+pip install natsort
+```
+
+
+## Running the Experiments
+Ensure that the environment is properly set up, then run:
+
+```bash  
+python -m flanders.main
+```
+
+To execute a single experiment with the default values in `conf/base.yaml`.
+
+To run custom experiments, you can override the defaults values like that:
+
+```bash
+python -m flanders.main dataset=income server.attack_fn=lie server.num_malicious=1
+```
+
+Finally, to run multiple custom experiments:
+
+```bash
+python -m flanders.main --multirun dataset=income,mnist,house,cifar server.attack_fn=gaussian,lie,fang,minmax server.num_malicious=0,1,2,3,4,5
+```
+
+
+## Expected Results
+
+By running;
+```bash
+python -m flanders.main --multirun dataset=income,mnist server.attack_fn=gaussian,lie,fang,minmax server.num_malicious=0,1,2,3,4,5,6,7,8,9
+```
+
+It will generate the results in `results/all_results.csv`. To generate the plots, use the notebook in `plotting/plots.ipynb`.
+
+Expected maximum accuracy achieved across different number of malicious clients and different attacks:
+![](_static/max_acc.jpg)
+
+Expected distribution of accuracy from round $L$ to the final round across all the experiments:
+![](_static/boxplot_Income.jpg)
+
+![](_static/boxplot_MNIST.jpg)
```

Comment:
@jafermarq here the MNIST boxplot is not coming up on GitHub. Any idea about why this happens?

Reply:
It seems adding a new line between the sentence and the line inserting `boxplot_Income.jpg` fixed it.